agent.go

```diff
@@ -463,52 +463,64 @@ func (ep *endpoint) deleteDriverInfoFromCluster() error {
 }
 
 func (ep *endpoint) addServiceInfoToCluster() error {
+	if ep.isAnonymous() && len(ep.myAliases) == 0 {
+		return nil
+	}
+
 	n := ep.getNetwork()
 	if !n.isClusterEligible() {
 		return nil
 	}
 
 	c := n.getController()
 	agent := c.getAgent()
-	if !ep.isAnonymous() && ep.Iface().Address() != nil {
```

Comment:
Thanks for catching this. I think it just got lost in the change. 
I am not sure if the address can ever be nil, but better to restore the original check.